pkg/backingstore/reconciler.go

```diff
@@ -989,24 +1002,47 @@ func (r *Reconciler) reconcileExistingPods(podsList *corev1.PodList) error {
 	return nil
 }
 
-func (r *Reconciler) reconcileMissingPods(podsList *corev1.PodList) error {
-	r.updatePodTemplate() // Adding Missing Agents
+func (r *Reconciler) needUpdate(pod *corev1.Pod) bool {
+	var c = &pod.Spec.Containers[0]
+	for _, name := range []string{"HTTP_PROXY", "HTTPS_PROXY", "NO_PROXY"} {
+		envVar := util.GetEnvVariable(&c.Env, name)
+		val, ok := os.LookupEnv(name)
+		if (envVar == nil && ok) || (envVar != nil && (!ok || envVar.Value != val)) {
+			r.Logger.Warnf("Change in Env varaibles detected: os(%s) container(%v)", val, envVar)
+			return true
+		}
+	}
+	if c.Image != r.NooBaa.Status.ActualImage {
+		r.Logger.Warnf("Change in Image detected: current image(%v) noobaa image(%v)", c.Image, r.NooBaa.Status.ActualImage)
+		return true
+	}
+	podSecrets := pod.Spec.ImagePullSecrets
+	noobaaSecret := r.NooBaa.Spec.ImagePullSecret
+	if noobaaSecret == nil {
+		sa := util.KubeObject(bundle.File_deploy_service_account_yaml).(*corev1.ServiceAccount)
+		sa.Name = pod.Spec.ServiceAccountName
+		sa.Namespace = options.Namespace
+		if util.KubeCheck(sa) && !reflect.DeepEqual(sa.ImagePullSecrets, pod.Spec.ImagePullSecrets) {
+			r.Logger.Warnf("Change in Image Pull Secrets detected: SA(%v) Spec(%v)", sa.ImagePullSecrets, pod.Spec.ImagePullSecrets)
+			return true
+		}
+	} else if noobaaSecret.Name != podSecrets[0].Name {
```

Comment:
maybe check if podSecrets contain noobaaSecret, would be safer

Reply:
as we currently replaces the whole podSecrets with an array of one secret - I think I will leave it with that - without adding extra risky complications: 
```
r.PodAgentTemplate.Spec.ImagePullSecrets =
			[]corev1.LocalObjectReference{*r.NooBaa.Spec.ImagePullSecret}
```
if we will change to adding in the future then we will handle